src/main/java/org/gridsuite/study/server/service/RemoteServicesInspector.java

```diff
@@ -106,15 +119,28 @@ public CompletableFuture<Boolean> isServerUp(RemoteServicesProperties.Service se
      * @implNote retrieve data in parallel to optimize requests time
      */
     @SuppressWarnings("unchecked") //.toArray(...) generics cause "Generic array creation) problem
-    public Map<String, JsonNode> getServicesInfo() {
-        final CompletableFuture<Entry<String, JsonNode>>[] resultsAsync = Arrays.stream(RemoteServiceName.values())
-                .parallel()
+    public Map<String, JsonNode> getServicesInfo(@Nullable FrontService viewFilter) throws PartialResultException {
```

Comment:
Update Javadoc ? (and no it's still not a filter lol, but you can say we filter the returned values depending of the front in the Javadoc)